diff -Naru libqalculate-3.3.0/libqalculate/util.cc libqalculate/libqalculate/util.cc
--- libqalculate-3.3.0/libqalculate/util.cc	2019-07-21 00:29:17.000000000 -0700
+++ libqalculate/libqalculate/util.cc	2019-08-20 13:24:45.421054061 -0700
@@ -898,8 +898,8 @@
 }
 
 void Thread::enableAsynchronousCancel() {
-	pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
-	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+	//pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
+	//pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
 }
 
 void *Thread::doRun(void *data) {
@@ -922,7 +922,8 @@
 
 bool Thread::cancel() {
 	if(!running) return true;
-	running = pthread_cancel(m_thread) != 0;
+	//running = pthread_cancel(m_thread) != 0;
+	running = pthread_kill(m_thread, 0) != 0;
 	return !running;
 }
 
